### DISCUSSION TITLE
Change user email address in a commit

### DISCUSSION THREAD

**Discussion by lmores** [community/community#37710] [Pull Requests] [answered]
Hello, I have two questions about email addresses inside commits.

  1. A few days ago I forked the repository of a public project, made same changes and opened a PR in the original repository.
    The PR was accepted and merged in the master branch (and other PRs have been merged in master since then).
    For the (only) commit in my PR I mistakenly used an email address that is not linked to my GitHub account.
    As a consequence, my contribution to the project is not displayed in my profile nor in the "contributors" section of the repository.
    I do not want to add the email address currently associated to the commit to the list of email addresses linked to my GitHub account.
    How can I fix this issue?
    Should I ask the maintainer of the project to do something to change the email address in my commit? How?
    
  2. Currently I have 3 email addresses linked to my GitHub account and I would like to get rid of two of them that I probably never used in my commits.
     However, I am not sure and before removing them I would like to check which of my commits (across all repositories) contain those email addresses.
     Is there a way to get the list of my commits that use those email addresses?
     
 Thank you a lot!

**Comment by airtower-luna** [comment, marked as answer]
> I do not want to add the email address currently associated to the commit to the list of email addresses linked to my GitHub account.
> How can I fix this issue?

Without adding the email address, you can't. *Before* the merge you could have amended the commit on your branch, but you can't do that in someone else's repository. See [Git Tools - Rewriting History](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) for the future.

> Should I ask the maintainer of the project to do something to change the email address in my commit? How?

*Theoretically* someone with the ability to force-push the default branch of that repository could still rewrite the history. In practice I'm pretty sure they'd reject such a request, because:

* *Everyone* with a copy of the repository would have to *manually* accept the modified history.
* It would change *all* commits added after yours (not the content, but the IDs), so it would cause a huge mess for anyone who's currently working on the code. People would need to very carefully rebase their branches (or cherry-pick their changes) onto the new history. Or if they don't, they'll reintroduce the unchanged commits, including yours. Basically, it'd be a huge mess.
* Inconvenience aside, if a change to repository history happens without a public announcement about the reasons it's very likely to make people very, very suspicious. I don't think either you or the maintainer would want to send out such an announcement. :sweat_smile:

**Reply by lmores** [reply]
Thank you for your answer. I knew it would be possible in theory using git, but I hoped there were also a practical solution. Amending the git history and force pushing to the master branch of that project is definitely not an option.

Therefore I added the email used in my commit to the list of emails linked to my account (and I verified that email address).
However, I still do not appear among the contributors of the project.
I also waited for more than 24h before writing this message in case "something" had to be triggered on the GitHub server side.
Do you know what I should do now?

**Reply by airtower-luna** [reply]
You can try [contact GitHub support](https://support.github.com/request). I've seen mentions around the forum that there are caches behind the scenes that support staff can flush/rebuild, where us other users can't help. Not sure if they'll let you request that for someone else's repository, but then it's just a cache either way.

**Reply by lmores** [reply]
I'm feeling pretty dumb, but I do not find out how to submit my request in the support page you linked. The "support categories" I see do not apply to my case and most of them redirect to documentation pages.

UPDATE: I submitted a ticket using the "Account or billing issues", let's see.

**Reply by lmores** [reply]
GitHub support answered my ticket: the "Contributors" section displays only the top 100 contributors and having just one commit in that repository is not enough.

As for question 2, it is possible to search for commits made using a specific email address using the `author-email:EMAIL` and `committer-email:EMAIL` search qualifiers, see [https://docs.github.com/en/search-github/searching-on-github/searching-commits#search-by-author-or-committer](https://docs.github.com/en/search-github/searching-on-github/searching-commits#search-by-author-or-committer).
